digdag-core/src/main/java/io/digdag/core/agent/InProcessTaskServerApi.java

```diff
@@ -7,11 +7,28 @@
 import io.digdag.spi.TaskQueueClient;
 import io.digdag.core.queue.TaskQueueServerManager;
 import io.digdag.core.workflow.WorkflowExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Comparator;
 import java.util.List;
+import java.util.function.ToLongFunction;
+import java.util.stream.Collectors;
 
 public class InProcessTaskServerApi
     implements TaskServerApi
 {
+    private static final Logger logger = LoggerFactory.getLogger(WorkflowExecutor.class);
+    private final ToLongFunction<TaskQueueLock> CONV_FUNC_FROM_TASK_QUEUE_LOCK_TO_INT = taskQueueLock -> {
+        try {
+            return WorkflowExecutor.parseTaskIdFromEncodedQueuedTaskName(taskQueueLock.getUniqueName());
+        }
+        catch (Throwable e) {
+            logger.warn("Failed to convert TaskQueueLock.uniqueName to integer. The `uniqueName` will be handled as 0", e);
+            return 0;
```

Comment:
Is there any possible case to execute this catch(){} code or just in case ?

Reply:
Basically for just-in-case. I was thinking of case that `uniqueName` unexpectedly contains non-digit string although I"m not sure it could happen.